frontends/web/src/api/account.ts

```diff
@@ -214,8 +214,10 @@ export interface ReceiveAddressList {
     addresses: IReceiveAddress[];
 }
 
-export const getReceiveAddressList = (code: AccountCode): Promise<ReceiveAddressList[]> => {
-    return apiGet(`account/${code}/receive-addresses`);
+export const getReceiveAddressList = (code: AccountCode) => {
```

Comment:
does it make sense to have some api functions return the promise directly while others return a function returning a promise? if you didn't change this function, would `useLoad(() => accountApi.getReceiveAddressList(code))` work? imo that would be a bit less magical and more expected.

i noticed because `accountApi.getReceiveAddressList(this.getAccount()!.code)()` looked a bit ugly.

Reply:
`accountApi.getReceiveAddressList(this.getAccount()!.code)()` looks indeed ugly, but will look nice once it  uses `useLoad()` hook, but that needs to be converted from class to FunctionalComponent. I wouldn't optimize for class.

We already use currying with `useSubscribe` https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/components/headerssync/headerssync.tsx#L33

and

https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/api/coins.ts#L29



Also it would be nice if it works with `useSync` currently is a combination of `useLoad` + `useSubscribe` which should just work with curried callbacks.

Reply:
okay but then please declare the return type on this line instead of inside the function body if possible